calico_kubernetes/calico_kubernetes.py

```diff
@@ -122,12 +129,20 @@ def _configure_interface(self):
         4) Assign the node's IP to the host end of the veth pair (required for
            compatibility with kube-proxy REDIRECT iptables rules).
         """
+        # Set up parameters
+        container_name = self.docker_id
+        workload_id = self._get_container_id(container_name)
+        container_pid = self._get_container_pid(container_name)
         container_ip = self._read_docker_ip()
-        container_id = self._get_container_id(self.docker_id)
+        container_ip = IPAddress(container_ip)
```

Comment:
_read_docker_ip() should return an IPAddress object so that you don't need to wrap it in one here.